chainerx_cc/chainerx/routines/math.cc

```diff
@@ -373,13 +390,29 @@ void DivideASImpl(const Array& x1, Scalar x2, const Array& out) {
         x1.device().backend().CallKernel<DivideASKernel>(x1, x2, out);
     }
 
-    BackwardBuilder bb{"divide_scalar", x1, out};
+    BackwardBuilder bb{"divide_array_scalar", x1, out};
     if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
         bt.Define([other = x2](BackwardContext& bctx) { bctx.input_grad() = *bctx.output_grad() / other; });
     }
     bb.Finalize();
 }
 
+void DivideSAImpl(Scalar x1, const Array& x2, const Array& out) {
+    {
+        NoBackpropModeScope scope{};
+        x2.device().backend().CallKernel<DivideSAKernel>(x1, x2, out);
+    }
+
+    BackwardBuilder bb{"divide_scalar_array", x2, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([other = x1, x2_tok = bb.RetainInput(0)](BackwardContext& bctx) {
+            const Array& x2 = bctx.GetRetainedInput(x2_tok);
+            bctx.input_grad() = -*bctx.output_grad() * other / (x2 * x2);
```

Comment:
How about using `chainerx::Square` for the performance?
```suggestion
            bctx.input_grad() = -*bctx.output_grad() * other / Square(x2);
```

Reply:
Thanks, fixed!